prow/bugzilla/client.go

```diff
@@ -272,6 +272,10 @@ func (c *client) AddPullRequestAsExternalBug(id int, org, repo string, num int)
 		return false, fmt.Errorf("failed to unmarshal JSONRPC response: %v", err)
 	}
 	if response.Error != nil {
+		if response.Error.Code == 100500 && strings.Contains(response.Error.Message, `duplicate key value violates unique constraint "ext_bz_bug_map_bug_id_idx"`) {
```

Comment:
I hate this string-matching pattern so much, but I assume there is not a better way to know.

Reply:
I don't know. It is possible that `100500` is a unique code for this type of error but I can't find any documentation about this (or any of the JSONRPC API in general) so this is just defensive so that we don't mask anything more serious.